src/vm/vm_memory/memory.rs

```diff
@@ -109,6 +109,25 @@ impl Memory {
             Err(memory_error) => Err(VirtualMachineError::MemoryError(memory_error)),
         }
     }
+
+    pub fn insert_relocatable(
+        &mut self,
+        key: &Relocatable,
+        val: &Relocatable,
+    ) -> Result<(), VirtualMachineError> {
+        self.insert(&MaybeRelocatable::from(*key), &MaybeRelocatable::from(*val))
+            .map_err(VirtualMachineError::MemoryError)
+    }
+
+    pub fn insert_integer(
+        &mut self,
+        key: &Relocatable,
+        val: &BigInt,
+    ) -> Result<(), VirtualMachineError> {
+        self.insert(&MaybeRelocatable::from(*key), &MaybeRelocatable::from(*val))
+            .map_err(VirtualMachineError::MemoryError)
+    }
```

Comment:
Maybe this can be reduced to a generic.

Reply:
Ill look into it

Reply:
Mergerd generic version from insert-generic branch, but kept insert_value wrapper to avoid the map_err after each call